### ISSUE TITLE
For posterity, confusion around custom sprites (ES6 Class) preUpdate

### ISSUE THREAD

**Issue by jbakerdev** [photonstorm/phaser#4814]
Leaving this here, hopefully will help someone out. Not sure if it's technically a bug:

After updating to 3.17 all the animations in my derived Sprites would only play the first frame, and then stop updating. In the past you use to need to use 

`this.scene.add.existing()` when using a custom Sprite. 
You also had to make sure your Sprite had a preUpdate method, which all of mine did. 

As of 3.17 however this is no longer necessary! If you had things setup the way I did with preUpdate methods in your derived Sprites, animations on those sprites will not work unless you manually call

`this.anims.update()` inside your custom preUpdate method!

TLDR; If your ES6 Sprites animation are not working, make sure you remove any preUpdate method you may have added OR manually call `this.anims.update()` inside preUpdate.

**Comment by rexrainbow** [comment]
Some Game objects have [preUpdate method](https://github.com/photonstorm/phaser/blob/master/src/gameobjects/sprite/Sprite.js#L115), just add `super.preUpdate()` in custom preUpdate method.

**Comment by photonstorm** [comment]
Just to add to this - nothing has changed since 3.17 with regard to `add.existing` etc - if you create a custom Game Object then it _needs_ to be added to the display list, and if it should update, as a Sprite does, then it needs adding to the update list as well.

You can either do this inside the Sprite constructor, i.e. `scene.sys.displayList.add(this)` or you could pass the created Sprite object to `add.existing(sprite)`. They both have the same end result.

If you've got an object that extends Sprite and needs its own logic inside of `preUpdate` then make sure to call `super.preUpdate(time, delta)` inside it.
